src/state/usePasscodeAuth/usePasscodeAuth.ts

```diff
@@ -69,6 +70,32 @@ export default function usePasscodeAuth() {
     [user]
   );
 
+  const updateRecordingRules = useCallback(
```

Comment:
I remember we talked about this duplicate code. Can you please remind me what we agreed on?

Reply:
Sure! Once we move from the `video-app-service` over to twilio functions (as part of [AHOYAPPS-964](https://issues.corp.twilio.com/browse/AHOYAPPS-964), which I will start this sprint), I'll be able to update `src/state/useFirebaseAuth` so that it consumes the new API. Since `usePasscodeAuth` and `useFirebaseAuth` will be consuming the same token server (with different authentication mechanisms), I'll be able to refactor those two hooks so that they share some logic. Additionally, I'd like to bring in the axios library, which will help us deal with errors a little better. 

Additionally a custom `useAxios` hook (or similar) will help to cut back on some of this duplicated code. Something like this:
`const {isRequesting, data, error} = useAxios('/token')`

In a nutshell, I will refactor much of `src/state/` when we move off video-app-service.

Reply:
Got it. Can you please add a small TODO above this line so I don't forget next time I look at this part of the code? 😄

Reply:
Well, I plan on refactoring more than this one function. I'll refactor a lot of stuff in `src/state`. Plus, it looks like this will be the next ticket I pick up. If I add a TODO it won't be there for long 🙂

Reply:
Ok, nvm then :)